stock_prices/fib.py

```diff
@@ -0,0 +1,39 @@
+# F(n) = F(n-1) + F(n-2)
+# F(0) = 0
+# F(1) = 1
+
+# add the two before it to get F(n)
+# Memoization! by adding cache
+cache = {}
+def F(n):
+  if n == 0:
+    return 0
+  if n == 1:
+    return 1
+  nth = F(n - 1) + F(n-2)
+
+  cache[n] = nth
+
+  return nth
+
+for i in range(40):
```

Comment:
good I'm glad you're looking at the relationship for between the terms. very helpful strategy.